src/components/sidebar/tabs/ModelLibrarySidebarTab.vue

```diff
@@ -183,10 +215,61 @@ watch(
   { deep: true }
 )
 
+const getDownloadLabel = ({
+  status,
+  filename
+}: Pick<ModelDownload, 'status' | 'filename'>) => {
+  switch (status) {
+    case 'cancelled':
+      return `${filename} [${t('electronFileDownload.cancelled')}]`
+    case 'paused':
+      return `${filename} [${t('electronFileDownload.paused')}]`
+    case 'completed':
+      return `${filename} [${t('electronFileDownload.completed')}]`
+    default:
+      return filename
+  }
+}
+
+const convertDownloadToModel = ({
+  url,
+  status,
+  filename,
+  savePath,
+  progress
+}: ModelDownload) => ({
+  key: url,
+  label: getDownloadLabel({ filename, status }),
+  leaf: true,
+  children: [],
+  progress: status === 'in_progress' ? progress : undefined,
+  data: {}
+})
+
 onMounted(async () => {
   if (settingStore.get('Comfy.ModelLibrary.AutoLoadAll')) {
     await modelStore.loadModels()
   }
+
+  if (isElectron()) {
+    const allDownloads: ModelDownload[] =
```

Comment:
Let's store the download tasks in a Pinia store, so that it becomes a global state accessible by all components. It would also help us keep the component decoupled from the downloading logic.